wgpu/src/backend/web.rs

```diff
@@ -1014,8 +1014,64 @@ impl crate::context::Context for Context {
             //Error: Tracing isn't supported on the Web target
         }
 
-        // TODO: non-guaranteed limits
         let mut mapped_desc = web_sys::GpuDeviceDescriptor::new();
+        log::info!("descriptor given: {:#?}", desc.limits);
+
+        // TODO: Migrate to a web_sys api.
+        // See https://github.com/rustwasm/wasm-bindgen/issues/3587
+        let limits_object = {
+            let object = js_sys::Object::new();
+
+            macro_rules! set_properties {
+                (($from:expr) => ($on:expr) : $(($js_ident:ident, $rs_ident:ident)),* $(,)?) => {
+                    $(
+                        ::js_sys::Reflect::set(
+                            &$on,
+                            &::wasm_bindgen::JsValue::from(stringify!($js_ident)),
+                            // TODO: Numbers may be bigger than u32!
+                            &::wasm_bindgen::JsValue::from($from.$rs_ident as u32)
```

Comment:
```suggestion
                            &::wasm_bindgen::JsValue::from($from.$rs_ident)
```

There's a u64 from impl?

Reply:
There is indeed, however it throws:
```
Non-JS-compatible Func Sigurature (i64 as parameter) can only be used for direct wasm to wasm dependencies
```
On the web. Perhaps we cast to `f64` instead and send that over?

Reply:
Yeah let's do that. Apparently u64 is somehow going through bigint?

Reply:
Yeah... want me to special-case it for only the parameters which are u64 or should I just do it for everything?

Reply:
Let's just do it for all of them - all a js number is is a f64 anywho.